OpenRA.Game/Map/MapCache.cs

```diff
@@ -358,6 +358,14 @@ IEnumerator IEnumerable.GetEnumerator()
 			return GetEnumerator();
 		}
 
+		public void Reset()
+		{
+			mapLocations.Clear();
+			generateMinimap.Clear();
```

Comment:
Careful. `generateMinimap` is used by the background loading thread and must protected by a lock.